contracts/src/tokens/points/RiverPoints.sol

```diff
@@ -65,6 +66,52 @@ contract RiverPoints is IERC20Metadata, IRiverPoints, OwnableBase, Facet {
         points = protocolFee * 3_000_000;
       }
     }
+
+    if (action == Action.CheckIn) {
+      (uint256 lastCheckIn, uint256 streak) = abi.decode(
+        data,
+        (uint256, uint256)
+      );
+      (points, ) = CheckIn.getPointsAndStreak(lastCheckIn, streak);
```

Comment:
Decode the address and read from storage?

Reply:
block.timestamp